docs/cert-manager.md

```diff
@@ -1,76 +1,448 @@
+
+```yaml
+# tenant-1-ca-certificate.yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: tenant-1-ca-certificate
+  namespace: tenant-1
+spec:
+  isCA: true
+  commonName: tenant-1-ca
+  secretName: tenant-1-ca-secret
+  duration: 70128h # 8y
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+  issuerRef:
+    name: selfsigned-root
+    kind: ClusterIssuer
+    group: cert-manager.io
+```
+
+```shell
+kubectl apply -f tenant-1-ca-certificate.yaml
+```
+
+## Deploy the tenant 
+
+### Create a certificate for Tenant
+
+Request Cert Manager issue a new TLS server certificate for MinIO that includes the following DNS domains:
+
+```shell
+minio.<namespace>
+minio.<namespace>.svc
+minio.<namespace>.svc.<cluster domain>
+*.<tenant-name>-hl.<namespace>.svc.<cluster domain>
+*.<namespace>.svc.<cluster domain>
+```
+
+> [!IMPORTANT]
+> Replace `<cluster domain>` with the actual values for your MinIO tenant.
+> * `cluster domain` is the internal root DNS domain assigned in your Kubernetes cluster. Typically this is `cluster.local`, check on your coredns
+> configuration for the correct value for your Kubernetes cluster. For example, using `kubectl get configmap coredns -n kube-system -oyaml | yq ".data"`.
+> The way the root DNS domain is managed can vary depending on the Kubernetes distribution (Openshift, Rancher, EKS, etc.)
+> * `tenant-name` is the name provided to your tenant in the `metadata.name` of the Tenant YAML. For this example it is `myminio`.
+> * `namespace` is the namespace where the tenant is created, the `metadata.namespace` notes that in the Tenant YAML. For this example it is `tenant-1`.
+
+Create a `Certificate` for the domains mentioned above:
 
 ```yaml
+# tenant-1-minio-certificate.yaml
 apiVersion: cert-manager.io/v1
 kind: Certificate
 metadata:
   name: tenant-certmanager-cert
-  namespace: minio-tenant
+  namespace: tenant-1
 spec:
   dnsNames:
-    - "*.tenant-certmanager.svc.cluster.local"
-    - "*.myminio.tenant-certmanager.svc.cluster.local"
-    - "*.myminio-hl.tenant-certmanager.svc.cluster.local"
-  secretName: tenant-certmanager-tls
+    - "minio.tenant-1"
+    - "minio.tenant-1.svc"
+    - '*.minio-tenant-1.svc.cluster.local'
+    - '*.minio.tenant-1.svc.cluster.local'
+    - '*.myminio-hl.tenant-1.svc.cluster.local'
+  secretName: myminio-tls
   issuerRef:
     name: tenant-certmanager-issuer
 ```
 
-Then it creates a new tenant including the new `tenant-certmanager-tls` secret in the `externalCertSecret` field.
+> [!TIP]
+> For this example, the Tenant name is `myminio`. We recommend naming the secret in the field `spec.secretName` as `<tenant name>-tls`, following the naming
+> convention MinIO Operator uses when creates certificates with Autocert enabled (`spec.requestAutoCert: true`).
+
+```shell
+kubectl apply -f tenant-1-minio-certificate.yaml
+```
+
+### Configure the tenant to use the certificate created by cert-manager 
+
+In the tenant spec, do the following:
```

Comment:
Perhaps suggest editing with `kubectl -n tenant-1 edit tenant/myminio`, or running the patch command suggested earlier.

Reply:
Same thing, a later patch will have adverse effects, we want to make sure tenant has the value in `spec.externalCertSecret` set from the moment is created